iceoryx_dds_gateway/include/ioxdds/gateway/iox2dds.hpp

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <functional>
+#include <memory>
+
+#include <iceoryx_posh/iceoryx_posh_types.hpp>
+#include <iceoryx_posh/popo/gateway_generic.hpp>
+#include <iceoryx_posh/popo/subscriber.hpp>
+#include <iceoryx_posh/runtime/posh_runtime.hpp>
+#include <iceoryx_utils/cxx/vector.hpp>
+#include <iceoryx_utils/internal/objectpool/objectpool.hpp>
+
+#include "ioxdds/dds/data_writer.hpp"
+#include "ioxdds/dds/dds_types.hpp"
+#include "ioxdds/gateway/channel.hpp"
+
+namespace iox
+{
+namespace gateway
+{
+namespace dds
+{
+// Configuration Parameters
+static constexpr uint32_t DISCOVERY_PERIOD_MS = 1000;
+static constexpr uint32_t FORWARDING_PERIOD_MS = 50;
+static constexpr uint32_t SUBSCRIBER_CACHE_SIZE = 128;
```

Comment:
Could you change them to `units::Duration`? Then we have the unit in the type rather than in the name.

Reply:
Fixed.